charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -121,3 +122,116 @@ service:
       exporters: [otlp]
 {{- end }}
 {{- end }}
+
+{{- define "opentelemetry-collector.agent.containerLogsConfig" -}}
+{{- if .Values.agentCollector.containerLogs.enabled }}
+receivers:
+  filelog:
+    include: [ /var/log/pods/*/*/*.log ]
+    {{- if not .Values.agentCollector.containerLogs.includeAgentLogs }}
+    exclude: [ /var/log/pods/{{ .Release.Namespace }}_{{ include "opentelemetry-collector.fullname" . }}-agent-*_*/{{ .Chart.Name }}/*.log ]
+    {{- end }}
+    start_at: beginning
+    include_file_path: true
+    include_file_name: false
+    operators:
+      {{- if eq .Values.agentCollector.containerLogs.containerRunTime "cri-o" }}
+      # Parse CRI-O format
+      - type: regex_parser
+        id: parser-crio
+        regex: '^(?P<time>[^ Z]+) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) (?P<log>.*)$'
+        output: extract_metadata_from_filepath
+        timestamp:
+          parse_from: time
+          layout_type: gotime
+          layout: '2006-01-02T15:04:05.000000000-07:00'
+      {{- end }}
+      {{- if eq .Values.agentCollector.containerLogs.containerRunTime "containerd" }}
+      # Parse CRI-Containerd format
+      - type: regex_parser
+        id: parser-containerd
+        regex: '^(?P<time>[^ ^Z]+Z) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) (?P<log>.*)$'
+        output: extract_metadata_from_filepath
+        timestamp:
+          parse_from: time
+          layout: '%Y-%m-%dT%H:%M:%S.%LZ'
+      {{- end }}
+      # Parse Docker format
+      {{- if eq .Values.agentCollector.containerLogs.containerRunTime "docker" }}
+      - type: json_parser
+        id: parser-docker
+        output: extract_metadata_from_filepath
+        timestamp:
+          parse_from: time
+          layout: '%Y-%m-%dT%H:%M:%S.%LZ'
+      {{- end }}
+      # Extract metadata from file path
+      - type: regex_parser
+        id: extract_metadata_from_filepath
+        regex: '^\/var\/log\/pods\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[^\/]+)\/(?P<container_name>[^\._]+)\/(?P<run_id>\d+)\.log$'
+        parse_from: $$attributes.file_path
+      # Move out attributes to Attributes
+      - type: metadata
+        attributes:
+          stream: 'EXPR($.stream)'
+          k8s.container.name: 'EXPR($.container_name)'
+          k8s.namespace.name: 'EXPR($.namespace)'
+          k8s.pod.name: 'EXPR($.pod_name)'
+          run_id: 'EXPR($.run_id)'
+          k8s.pod.uid: 'EXPR($.uid)'
+        resource:
+          k8s.pod.uid: 'EXPR($.uid)'
+      # Clean up log record
+      - type: restructure
+        id: clean-up-log-record
+        ops:
+          - remove: logtag
+          - remove: stream
+          - remove: container_name
+          - remove: namespace
+          - remove: pod_name
+          - remove: run_id
+          - remove: uid
+{{- if .Values.agentCollector.containerLogs.enrichK8sMetadata }}
+processors:
+  k8s_tagger:
```

Comment:
`groupbyattrs` processot is required to use before `k8s_tagger` as `k8s_tagger` uses resource labels

```
  groupbyattrs:
    keys:
      - k8s.container.name
      - k8s.namespace.name
      - k8s.pod.name
      - run_id
      - k8s.pod.uid
```

Reply:
`groupbyattrs` will have a performance impact. Probably we have to do it now, but for the future I think it would be better if `filelog` allowed placing some attributes directly into the resource and we use that capability in k8s collection. I opened an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2725

Reply:
I am putting it in resource label, no? 
```
      - type: metadata
        attributes:
          stream: 'EXPR($.stream)'
          k8s.container.name: 'EXPR($.container_name)'
          k8s.namespace.name: 'EXPR($.namespace)'
          k8s.pod.name: 'EXPR($.pod_name)'
          run_id: 'EXPR($.run_id)'
          k8s.pod.uid: 'EXPR($.uid)'
        resource:
          k8s.pod.uid: 'EXPR($.uid)'
```

Reply:
I confirm enrichment works without using groupbyattrs. 
```
2021-03-20T00:05:40.172Z        INFO    loggingexporter/logging_exporter.go:471 LogsExporter    {"#logs": 1}
2021-03-20T00:05:40.173Z        DEBUG   loggingexporter/logging_exporter.go:498 ResourceLog #0
Resource labels:
     -> k8spoduid: STRING(eda0a956-99ec-49cb-a954-d733e2f881dd)
     -> k8s.pod.annotations.splunk.com/index: STRING(test)
     -> k8s.pod.name: STRING(loggen-zjklf)
     -> k8s.namespace.name: STRING(default)
     -> k8s.pod.startTime: STRING(2021-03-20 00:03:32 +0000 UTC)
     -> k8s.pod.uid: STRING(eda0a956-99ec-49cb-a954-d733e2f881dd)
     -> k8s.node.name: STRING(ip-192-168-82-91.us-west-1.compute.internal)
     -> k8s.pod.labels.hello: STRING(loggen)
     -> k8s.pod.labels.app: STRING(loggen)
InstrumentationLibraryLogs #0
InstrumentationLibrary  
LogRecord #0
Timestamp: 1616198740006040006
Severity: Undefined
ShortName: 
Body: {
     -> log: STRING(num: 127 | loggen-zjklf | 2021-03-20T00:05:40.005870 | rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrr
)
}
Attributes:
     -> k8s.pod.uid: STRING(eda0a956-99ec-49cb-a954-d733e2f881dd)
     -> run_id: STRING(0)
     -> stream: STRING(stdout)
     -> k8s.container.name: STRING(loggen)
     -> k8s.namespace.name: STRING(default)
     -> k8s.pod.name: STRING(loggen-zjklf)
```

@tigrannajaryan 
However, when i ingest this log to splunk, I only see metadata in `Attributes`, not in `Resources`
![image](https://user-images.githubusercontent.com/12387289/111853046-8de66e80-88d6-11eb-9971-3385c49ee1fb.png)
What should I do to ingest data in `Attributes` ?

Reply:
@rockb1017 see if @atoulme can help, I have never used the HEC exporter. My guess is that HEC exporter does not handle Resource attributes, but I may be wrong. If that's the case then it is a bug in HEC exporter.